### PR TITLE
Hotfix: add default price label when creating issue

### LINKED ISSUE #482: Hotfix: Price label not generated when Default labels are set
image

Bot should calculate and set a price label if both time and priority are set, however it seems the event to calculate price cant be triggered by the bot itself in some instances

### PR DESCRIPTION
**Pull request by wannacfuture**
Resolves #482 

QA example: https://github.com/wannacfuture/Battleship/issues/29

### CODE REVIEW

**Review by pavlovcik — DISMISSED** [review]
The code checks out but @0xcodercrane is this the only condition the bot needs in order to generate a payment permit? A `Price` label?

File: src/handlers/comment/handlers/index.ts [hunk 1]
```diff
@@ -75,6 +76,16 @@ export const issueCreatedCallback = async (): Promise<void> => {
...
+      const exist = await getLabel(targetPriceLabel);
+      if (!exist) {
+        await createLabel(targetPriceLabel, "price");
+      }
+      await addLabelToIssue(targetPriceLabel);
+    }
+
     return;
   } catch (err: unknown) {
     return await addCommentToIssue(`Error: ${err}`, issue.number);
```
Comment (0xcodercrane) [inline src/handlers/comment/handlers/index.ts:91]:
```suggestion
    await addCommentToIssue(`Error: ${err}`, issue.number);
```

File: src/handlers/comment/handlers/index.ts [hunk 2]
```diff
@@ -75,6 +76,16 @@ export const issueCreatedCallback = async (): Promise<void> => {
...
+    const targetPriceLabel = getTargetPriceLabel(timeLabelConfigs[0].name, priorityLabelConfigs[0].name);
+    if (targetPriceLabel) {
+      const exist = await getLabel(targetPriceLabel);
+      if (!exist) {
+        await createLabel(targetPriceLabel, "price");
+      }
+      await addLabelToIssue(targetPriceLabel);
+    }
+
     return;
```
Comment (0xcodercrane) [inline src/handlers/comment/handlers/index.ts:89]:
```suggestion
```

File: src/handlers/comment/handlers/index.ts [hunk 3]
```diff
@@ -75,6 +76,16 @@ export const issueCreatedCallback = async (): Promise<void> => {
...
+
+    const targetPriceLabel = getTargetPriceLabel(timeLabelConfigs[0].name, priorityLabelConfigs[0].name);
+    if (targetPriceLabel) {
+      const exist = await getLabel(targetPriceLabel);
+      if (!exist) {
+        await createLabel(targetPriceLabel, "price");
+      }
+      await addLabelToIssue(targetPriceLabel);
+    }
+
```
Comment (0xcodercrane) [inline src/handlers/comment/handlers/index.ts:88]: would be better to remove duplication use of `timeLabelConfigs[0].name` and `priorityLabelConfigs[0].name`

**Review by 0xcodercrane — CHANGES_REQUESTED** [review]

**Comment by 0xcodercrane** [comment]
Holding off the PR until the exploit issue resolved

**Comment by pavlovcik** [comment]
> Holding off the PR until the exploit issue fully resolved

Can you link back an open pull request working on resolving the exploit? It seems like this pull request was forgotten about and its difficult for me to audit the progress.

**Comment by 0xcodercrane** [comment]
> > Holding off the PR until the exploit issue fully resolved
> 
> Can you link back an open pull request working on resolving the exploit? It seems like this pull request was forgotten about and its difficult for me to audit the progress.

https://github.com/ubiquity/ubiquibot/commit/f4e5c65259d3bc9d7d6a1fc3468ca2b19733a624

**Review by Draeieg — APPROVED** [review]

**Comment by pavlovcik** [comment]
Can we merge this @0xcodercrane?

**Comment by 0xcodercrane** [comment]
> Can we merge this @0xcodercrane? The merge conflict is getting complicated.

Yes. let me handle it

**Review by 0xcodercrane — APPROVED** [review]

**Pull request merged**